backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -995,6 +1027,71 @@ public void legacyRbacUpdateEventTypesLinkedToEndpoint(final SecurityContext sec
 
     private void internalUpdateEventTypesLinkedToEndpoint(final SecurityContext securityContext, final UUID endpointId, final Set<UUID> eventTypeIds) {
         final String orgId = getOrgId(securityContext);
-        endpointEventTypeRepository.updateEventTypesLinkedToEndpoint(endpointId, eventTypeIds, orgId);
+        Endpoint updatedEndpoint =  endpointEventTypeRepository.updateEventTypesLinkedToEndpoint(endpointId, eventTypeIds, orgId);
+
+        // Sync behavior group model
+
+        // delete endpoint from existing behavior group
+        List<BehaviorGroup> behaviorGroupsLinkedToThisEndpoint = behaviorGroupRepository.findBehaviorGroupsByEndpointId(orgId, endpointId);
+        for (BehaviorGroup behaviorGroup : behaviorGroupsLinkedToThisEndpoint) {
+            if (behaviorGroup.getActions().size() == 1) {
+                behaviorGroupRepository.delete(orgId, behaviorGroup.getId());
+            } else {
+                behaviorGroupRepository.deleteEndpointFromBehaviorGroup(behaviorGroup.getId(), endpointId, orgId);
+            }
+        }
+
+        // Create or update relevant behavior groups
+        createOrUpdateLinkedBehaviorGroup(eventTypeIds, endpointId, updatedEndpoint.getName(), orgId);
+    }
+
+    private void createOrUpdateLinkedBehaviorGroup(Set<UUID> eventTypeIds, UUID endpointId, String endpointName, String orgId) {
+        String behaviorGroupName = String.format("Integration \"%s\" behaviour group", endpointName);
```

Comment:
Behavior group names have to be unique in prod (the feature toggle is on). Is there a chance this would collide with an existing name?